roles/bootstrap/tasks/application-credential-cleanup.yml

```diff
@@ -0,0 +1,48 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Assert cifmw_bootstrap_cloud_name is defined
+  ansible.builtin.assert:
+    that:
+      - cifmw_bootstrap_cloud_name is defined
+    msg: "Cloud name is mandatory"
+    quiet: true
+
+- name: Clear application credential from cloud_secrets
+  ansible.utils.update_fact:
+    updates:
+      - path: "cloud_secrets.{{cifmw_bootstrap_cloud_name}}.auth_type"
+        value: password
+      - path: "cloud_secrets.{{cifmw_bootstrap_cloud_name}}.application_credential_id"
+        value:
+      - path: "cloud_secrets.{{cifmw_bootstrap_cloud_name}}.application_credential_secret"
+        value:
+  register: updated_cloud_secrets
+
+- name: Replace cloud_secrets with updated
+  set_fact:
+    cloud_secrets: "{{ updated_cloud_secrets.cloud_secrets }}"
+
+- name: Rewrite clouds.yaml file with original credential
+  ansible.builtin.template:
+    src: clouds.yaml.j2
+    dest: "{{ cifmw_bootstrap_clouds_yaml }}"
+    mode: "0600"
+
+- name: Delete application credential
+  application_credential:
```

Comment:
Actually my intention is that this use the application_credential module I'm proposing to ci-framework:

https://github.com/openstack-k8s-operators/ci-framework/pull/1296

This is identical to the one proposed to openstack.cloud but it will be available much quicker, and it won't be in the openstack.cloud namespace. We can swap out later once openstack.cloud.application_credential is consumable, which could be a while

Reply:
ah, ok I did not notice the depends on.

A nit, but if you do another pass. We want to clone the ci-framework manually to the controller when running with the "local"/"injected" tags for lab environment. Similar to how I copy over the ci-bootstrap repo[1] - probably just use ansible git module to clone ci-framework in that block - not sure if we would need to do something to inform ansible about the roles/modules path?. (In real CI the repos are cloned by the prepare-workspace[2] role.)

[1] https://github.com/openstack-k8s-operators/ci-bootstrap/blob/staging/roles/bootstrap/tasks/controller-pre.yml#L38
[2] https://opendev.org/zuul/zuul-jobs/src/branch/master/roles/prepare-workspace